src/kvstore/comm.h

```diff
@@ -693,9 +729,41 @@ class CommDevice : public Comm {
           min_size = size;
         }
       }
-      buf.merged = NDArray(s, ctx, false, type);
+      stage.merged = NDArray(s, ctx, false, type);
       ctx_info[ctx.dev_id].second += s.Size();
     }
+    } else {
```

Comment:
What's the impact of this update on older devices/architectures?

Reply:
It'll avoid using QPIs.